File: src/main/java/com/lessmarkup/interfaces/structure/NotificationProvider.java
package com.lessmarkup.interfaces.structure;

import com.lessmarkup.interfaces.data.DomainModel;
import java.util.OptionalLong;

public interface NotificationProvider {
    String getTitle();
    String getTooltip();
    String getIcon();
    int getValueChange(OptionalLong fromVersion, OptionalLong toVersion, DomainModel domainModel);
}

File: src/main/scala/com/lessmarkup/interfaces/structure/NotificationProvider.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.structure

import com.lessmarkup.interfaces.data.DomainModel

trait NotificationProviderFactory

trait NotificationProvider {
  def getTitle: String

  def getTooltip: String

  def getIcon: String

  def getValueChange(fromVersion: Option[Long], toVersion: Option[Long], domainModel: DomainModel): Int
}